.circleci/config.yml

```diff
@@ -70,6 +72,58 @@ jobs:
             - ~/.cargo/registry/cache/
             - ~/.cargo/git/db/
             - target/
+  bats:
+    parameters:
+      os:
+        type: string
+    executor: << parameters.os >>
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - cargo-build-{{ .Environment.MANY_RS_CACHE_VERSION }}-{{ arch }}-{{ checksum "Cargo.lock" }}
+      - rust/install:
+          version: nightly
+      # TODO: Make a release with `many` CLI and install the release instead
+      - run:
+          name: install `many` CLI
+          command: cargo install --force --git https://github.com/liftedinit/many-rs many
+      - run:
+          name: install `ledger` CLI
+          command: cargo install --force --path src/ledger ledger
+      - run:
+          name: install `kvstore` CLI
+          command: cargo install --force --path src/kvstore kvstore
+      - run:
+          name: install cbor-diag
+          command: cargo install --force cbor-diag-cli
+      - bats/install
+      - run:
+          name: running BATs tests
+          command: bats *
+          working_directory: ./tests/e2e/
+  dev_release_artifact:
+    parameters:
+      os:
+        type: string
+    executor: << parameters.os >>
+    steps:
+      - checkout
+      - detect/init
+      - rust/install:
+          version: nightly
+      - rust/build:
+          release: true
+          with_cache: false
+      - run: mkdir -p artifacts
+      - run:
+          name: creating release archive
+          command: file target/release/* | grep 'executable\|shared object' | cut -d ':' -f 1 | xargs tar czvf artifacts/many-rs_PR_${CIRCLE_PR_NUMBER}_${CIRCLE_SHA1}_${OSD_ID}_${OSD_VERSION}.tar.gz
```

Comment:
nit: For some reason, we have some `.so` in the archive now.

Reply:
Let me check.